src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -145,6 +145,8 @@ private void SetGlobalErrorIfTrue(bool arg)
                     methodCompiler.CompileSynthesizedMethods(additionalTypes, diagnostics);
                 }
 
+                compilation.MethodGroupConversionCacheFrameManager.AssignFrameNames(moduleBeingBuiltOpt, diagnostics);
```

Comment:
Add a comment why the container types can't be named upfront and have to have their naming deferred to here.